apps/namadillo/src/slices/proposals/functions.ts

```diff
@@ -180,276 +172,141 @@ const decodeProposalType = async (
-  chain: Chain,
-  id: bigint,
-  startEpoch: bigint,
-  endEpoch: bigint
-): Promise<
-  {
-    [VT in VoteType]: BigNumber;
-  } & {
-    totalVotingPower: BigNumber;
-    status: ProposalStatus;
-  }
-> => {
-  const currentEpoch = await fetchCurrentEpoch(chain);
-
-  const epoch = endEpoch < currentEpoch ? endEpoch : currentEpoch;
-
-  if (NAMADA_INTERFACE_NO_INDEXER) {
-    const status: ProposalStatus =
-      startEpoch > currentEpoch ? "pending"
-      : endEpoch > currentEpoch ? "ongoing"
-      : id % BigInt(2) === BigInt(0) ? "passed"
-      : "rejected";
-
-    return {
-      status,
-      yay: BigNumber(300),
-      nay: BigNumber(250),
-      abstain: BigNumber(125),
-      totalVotingPower: BigNumber(1000),
-    };
-  }
-
-  const { rpc } = chain;
-  const query = new Query(rpc);
-
-  const [passed, yayPower, nayPower, abstainPower, totalVotingPower]: [
-    boolean,
-    string,
-    string,
-    string,
-    string,
-  ] = await query.query_proposal_result(id, epoch);
-
-  const yayPowerAsBigNumber = BigNumber(yayPower);
-  if (yayPowerAsBigNumber.isNaN()) {
-    throw new Error("couldn't parse yay power as BigNumber");
-  }
-
-  const nayPowerAsBigNumber = BigNumber(nayPower);
-  if (nayPowerAsBigNumber.isNaN()) {
-    throw new Error("couldn't parse nay power as BigNumber");
-  }
+export const fetchProposalVotes = async (id: bigint): Promise<Vote[]> => {
+  const api = new DefaultApi();
 
-  const abstainPowerAsBigNumber = BigNumber(abstainPower);
-  if (abstainPowerAsBigNumber.isNaN()) {
-    throw new Error("couldn't parse abstain power as BigNumber");
-  }
+  const response = await api.apiV1GovProposalIdVotesGet(Number(id));
 
-  const totalVotingPowerAsBigNumber = BigNumber(totalVotingPower);
-  if (totalVotingPowerAsBigNumber.isNaN()) {
-    throw new Error("couldn't parse total voting power as BigNumber");
-  }
+  // TODO: This is only needed for votes breakdown, check if it's still relevant
+  const votingPower: [string, BigNumber][] = [
+    ["_", BigNumber(0)], // TODO: return voting power
+  ];
 
-  const status: ProposalStatus =
-    startEpoch > currentEpoch ? "pending"
-    : endEpoch > currentEpoch ? "ongoing"
-    : passed ? "passed"
-    : "rejected";
+  const votes: Vote[] = response.data.data.map((vote) => ({
+    address: vote.voterAddress,
+    voteType: vote.vote,
+    votingPower,
+    isValidator: false,
+  }));
 
-  return {
-    status,
-    yay: yayPowerAsBigNumber,
-    nay: nayPowerAsBigNumber,
-    abstain: abstainPowerAsBigNumber,
-    totalVotingPower: totalVotingPowerAsBigNumber,
-  };
+  return votes;
 };
 
 export const fetchProposalVoted = async (
-  chain: Chain,
   id: bigint,
   account: Account
 ): Promise<boolean> => {
-  const votes = await fetchProposalVotes(chain, id);
+  const votes = await fetchProposalVotes(id);
   return votes.some((vote) => vote.address === account.address);
 };
 
 export const fetchVotedProposalIds = async (
-  chain: Chain,
+  _chain: Chain,
```

Comment:
I think it's safe to just remove this and update where applicable, but we can do that later!